build/Build.cs

```diff
@@ -456,7 +458,7 @@ public Build()
         
         Target UpdateCalamariVersionOnOctopusServer =>
             _ =>
-                _.Requires(() => SetOctopusServerVersion)
+                _.OnlyWhenStatic(() => SetOctopusServerVersion)
```

Comment:
I think the intention of this line is to only run this target when we set `SetOctopuServerVersion` to true. `Requires` is the wrong method to use here, since if `SetOctopuServerVersion` is false, it just prevents the whole build from running, rather than skipping this particular target.